tests/phpunit/tests/block-supports/dimensions.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+/**
+ * @group block-supports
+ *
+ * @covers ::wp_apply_dimensions_support
+ */
+
+class WP_Block_Supports_Dimensions_Test extends WP_UnitTestCase {
+	/**
+	 * @var string|null
+	 */
+	private $test_block_name;
+
+	public function set_up() {
+		parent::set_up();
+		$this->test_block_name = null;
+	}
+
+	public function tear_down() {
+		unregister_block_type( $this->test_block_name );
+		$this->test_block_name = null;
```

Comment:
A couple of questions:

1. Does the parent `::tear_down()` method not unregister block types?
2. Is it necessary to set new block names for each test, or can we make `$test_block_name` a `static` property and set it in `public static set_up_before_class()` instead?

Reply:
Good questions!

1. It doesn't appear to. The teardown methods mostly appear to tear down database related things, so for tests that include registering blocks, it appears that many of the other similar test files register / unregister in setup / teardown methods.
2. I'm not sure if it's strictly necessary to have unique names for each test block, but since it's testing registering a block with unique dimensions support params, I like the idea of it also having a unique block name at registration time, in that we know it won't be re-using an existing test block name. Also, the patterns in this test were copied from the other block supports test files (`border.php`, `colors.php`, `spacing.php`, `typography.php`), and I think it helps readability a bit if we stick to a consistent convention across those files in terms of registering / unregistering blocks with block supports.

So from my perspective, I think the data provider helped simplify the readability of the tests, but personally I'd probably lean toward sticking with the register/unregister convention from the other block supports tests.